src/generic/PermGroups.jl

```diff
@@ -525,39 +531,64 @@ end
 #
 ###############################################################################
 
-Base.start(A::AllPerms{<:Integer}) = 0
-
-function Base.next(A::AllPerms, count)
-    count = nextperm(A, count)
-    return A.elts, count
+function Base.iterate(A::AllPerms{<: Integer})
+   return A.elts, 1
 end
 
-Base.done(A::AllPerms, count) = count >= A.all
-Base.eltype(::Type{AllPerms{T}}) where T<:Integer = perm{T}
-Base.length(A::AllPerms) = A.all
-
-function nextperm(A::AllPerms{<:Integer}, count)
-   if count == 0
-        return count+1
+function Base.iterate(A::AllPerms{<: Integer}, count)
+   if count >= A.all
+     return nothing
    end
 
    k = 0
    n = 1
 
-   @inbounds while true
```

Comment:
why without `@inbounds`? returning A.elts messes things up?

Reply:
My mistake. I was just trying to get the iterators running with the new iterator interface. I think it would be best if we merge it first and then you could try to clean it up. Does this sound OK for you?

Reply:
sure. I just had a quick look on my phone on the changes, things look ok;
I will have a closer look at this in the coming week. Merge if You think it's ready.